### ISSUE TITLE
TypedDict should be incompatible with Protocol

### ISSUE THREAD

**Issue by jfmcoronel** [microsoft/pyright#2415]
**Describe the bug**

mypy 0.910 checking flags an error for the code below which passes a `TypedDict` value into a `Protocol`-expecting function; it erroneously passes Pyright 1.1.177 checking.

**To reproduce**

Run the code below and verify that it erroneously passes Pyright 1.1.177 checking.

**Expected behavior**

Pyright should flag `print_name(my_typed_dict)` with an error message similar to `error: Argument 1 to "print_name" has incompatible type "B"; expected "HasName"` of mypy 0.910.

**Screenshots or code**
```py
# pyright: strict
from dataclasses import dataclass
from typing import Protocol, TypedDict

class HasName(Protocol):
    name: str

@dataclass
class A:
    name: str

class B(TypedDict):
    name: str

def print_name(x: HasName):
    print(x.name)

my_typed_dict: B = {'name': 'my name'}

print_name(A('my name'))
print_name(my_typed_dict)
```

**VS Code extension or command-line**

Checked via Pyright command-line tool, coc.nvim of Vim, and Pylance of VS Code.

**Additional context**

`x.name` as ensured by `Protocol` is incompatible with `x['name']` as needed by `TypedDict`. I don't see anything in [PEP 544](https://www.python.org/dev/peps/pep-0544/#protocol-members) about this case in particular.

**Comment by erictraut** [comment]
Thanks for the bug report. I agree this should be flagged as a type error.

**Comment by erictraut** [comment]
This will be fixed in the next release of pyright and pylance.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.178, which I just published. It will also be included in the next release of pylance.
